pkg/entities/vault.go

```diff
@@ -48,3 +51,74 @@ func (e *Entity) CreateConfigThreshold(req *request.CreateConfigThresholdRequest
 		Threshold: req.Threshold,
 	})
 }
+
+func (e *Entity) AddTreasurerToVault(req *request.AddTreasurerToVaultRequest) (*model.Treasurer, error) {
+	addTreasurerReq, err := e.repo.TreasurerRequest.GetById(req.RequestId)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return nil, fmt.Errorf("request not exist")
+		}
+		e.log.Fields(logger.Fields{"req": req}).Errorf(err, "[entity.AddTreasurerToVault] - e.repo.TreasurerRequest.GetById failed")
+		return nil, err
+	}
+
+	// get vault from name and guild id
```

Comment:
=> getById, where Name?

Reply:
Forgot to edit the comments, get vault from request id only